libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -183,6 +182,17 @@ void AP_UAVCAN::init(uint8_t driver_index)
 
     hw_version.major = AP_UAVCAN_HW_VERS_MAJOR;
     hw_version.minor = AP_UAVCAN_HW_VERS_MINOR;
+
+    uint8_t len = hw_version.unique_id.capacity();
+    uint8_t *unique_id = new uint8_t[len];
+
+    if (unique_id) {
+    hal.util->get_system_id_unformatted(unique_id, len);
+    uavcan::copy(unique_id, unique_id + len, hw_version.unique_id.begin());
+    } else {
+        AP_HAL::panic("UAVCAN: Failed to allocate Unique ID");
```

Comment:
this panic is related to failed memory allocation, which is more serious issue to have during startup.

Reply:
Where else do we panic for a failed memory allocation on startup? This is an optional operation, so I don't think we even need to notify the user, let alone crashing because of it.

Feel free to hear other people's opinion.

Reply:
@tridge would be perfect person to comment on this. But my opinion is if memory allocation fails at any stage that's not an isolated event just for UAVCAN Server part, it may be caused by issues at some other location or might become cause issue at other locations.

Reply:
the panic is gone now
I also think many of the panics in ArduPilot should be replaced by the sensor_error loop, so at least the user knows what has happened